moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -513,9 +538,75 @@ KinematicState Servo::getNextJointState(const ServoInput& command)
     }
   }
 
+  // remove old commands
+  cur_time = node_->now();
+  while (!committed_commands_.empty() &&
+         committed_commands_.front().time < (cur_time - rclcpp::Duration::from_seconds(2 * servo_params_.latency)))
+  {
+    committed_commands_.pop_front();
+  }
+
+  // add next committed command
+  target_state.time = node_->now() + rclcpp::Duration::from_seconds(servo_params_.latency);
+  committed_commands_.push_back(target_state);
+
+  // add end command stop point in case of large delay
+  auto dt = 2 * (servo_params_.latency);
+  auto end_state = target_state;
```

Comment:
So this is like an added halt state, to help with JTC interpolation? That makes sense. I bet that's the main reason for the smoother performance.

I would tweak the wording a little bit here. Can discuss later, if my understanding is correct.

Reply:
We might need to think more about this. There is a filteredHalt() function for coming to a stop, gently. This would be an abrupt halt.

I'm also not sure it's necessary. If the JTC position interface is active, it will automatically stop.

Further realizing that this code path should never be executed anyway, unless the latency is extreme.